activerecord/test/cases/adapters/mysql2/nested_deadlock_test.rb

```diff
@@ -115,5 +131,13 @@ class Sample < ActiveRecord::Base
       assert_equal 1, deadlocks
       assert_equal [10, 10], Sample.pluck(:value)
     end
+
+    private
+      def flunk_unless_savepoint_transaction
+        current_transaction = Sample.connection.current_transaction
+        unless current_transaction.is_a?(ActiveRecord::ConnectionAdapters::SavepointTransaction)
+          flunk("current transaction is not a savepoint transaction")
+        end
+      end
```

Comment:
Could this also be suitably named `assert_in_savepoint_transaction` or `assert_current_transaction_is_savepoint_transaction`?

Reply:
Definitely, but do you think it's still reasonable to use `flunk`? I added an assertion first, but I feel like conceptually `assert` should be used for actual expectations of the test and `flunk` is more suitable to fail test in case if test setup is wrong

Reply:
I hadn't considered that distinction!  I'm not sure if Rails makes that distinction elsewhere, but it sounds reasonable to me.